src/extensions/anyone_can_pay/mod.rs

```diff
@@ -0,0 +1,236 @@
+        _net_ty: NetworkType,
+        config: HashMap<String, DeployedScriptConfig>,
+    ) -> Self {
+        ACPExtension {
+            store,
+            indexer,
+            _net_ty,
+            config,
+        }
+    }
+
+    fn get_live_cell_by_out_point(&self, out_point: &packed::OutPoint) -> Result<DetailedLiveCell> {
+        self.indexer
+            .get_detailed_live_cell(out_point)?
+            .ok_or_else(|| {
+                ACPExtensionError::CannotGetLiveCellByOutPoint {
+                    tx_hash: hex::encode(out_point.tx_hash().as_slice()),
+                    index: out_point.index().unpack(),
+                }
+                .into()
+            })
+    }
+
+    fn is_acp_cell(&self, cell_output: &packed::CellOutput) -> bool {
+        let script = cell_output.lock();
+        let config = self
+            .config
+            .get(ACP)
+            .unwrap_or_else(|| panic!("ACP extension config is empty"));
+
+        if script.code_hash() == config.script.code_hash()
+            && script.hash_type() == config.script.hash_type()
+        {
+            return true;
+        }
+
+        false
+    }
+
+    fn get_acp_pubkey_hash(&self, lock_args: &packed::Bytes) -> H160 {
+        let hash: Vec<u8> = lock_args.unpack();
+        H160::from_slice(&hash[0..20]).unwrap()
+    }
+
+    fn store_acp_cells(
+        &self,
+        acp_map: ACPMap,
+        block_num: BlockNumber,
+        block_hash: &packed::Byte32,
+        is_reverse: bool,
+    ) -> Result<()> {
+        let mut batch = self.get_batch()?;
+
+        for (key, mut val) in acp_map.0.clone().into_iter() {
+            if is_reverse {
+                val.reverse();
+            }
+
+            let addr_key = Key::CkbAddress(&key);
+            let mut out_points =
+                self.store
+                    .get(&addr_key.clone().into_vec())?
+                    .map_or_else(Vec::new, |bytes| {
+                        packed::OutPointVec::from_slice(&bytes)
+                            .unwrap()
+                            .into_iter()
+                            .collect()
+                    });
+
+            for removed in val.removed.into_iter() {
+                if !out_points.contains(&removed) {
+                    return Err(ACPExtensionError::MissingACPCell {
+                        tx_hash: hex::encode(removed.tx_hash().as_slice()),
+                        index: removed.index().unpack(),
+                    }
+                    .into());
+                }
+
+                remove_item(&mut out_points, &removed);
+            }
+
+            for added in val.added.into_iter() {
+                out_points.push(added);
+            }
+
+            batch.put_kv(addr_key, Value::ACPCells(out_points.pack()))?;
+        }
+
+        batch.put_kv(
+            Key::Block(block_num, block_hash),
+            Value::RollbackData(rlp::encode(&acp_map)),
+        )?;
+        batch.commit()?;
+
+        Ok(())
+    }
```

Comment:
If an acp cell is create and then consumed in the same block, then it will trigger two bugs:
1. Panic a MissingACPCell error
2. Even if you skip the first obstacle, this consumed acp will append to live cells

To solve this problem, you should remove the intersection acps from removed and added groups.

Reply:
LGTM